lineairdb_field.h

```diff
@@ -0,0 +1,55 @@
+#include <string>
+#include "my_inttypes.h"
+
+/**
+ * @brief This class is responsible for the translation between
+ * MySQL Field value and LineairDB Field.
+ * @details
+ * LineairDB field consists of the following 3 information:
+ *  header1     header2
+ * [byteSize][valueLength][value]
+ * header info
+ * - byteSize: number of bytes of `valueLength`
+ *             always 1 byte, byteSize = UCHAR_MAX if valueLength = 0
+ * - valueLength: length of value
+ *                max 8 bytes
+ * value: MySQL value shown to users
+ *        max 4294967295 = sizeof(LONGBLOB) bytes
+ * Each row consists of multiple fields.
+ * First field stores null flags.
+ */
+class Lineairdb_Field
+{
+public:
+  std::string convert_numeric_to_bytes (const size_t num);
+  template <typename BYTE_TYPE>
+  size_t convert_bytes_to_numeric(const BYTE_TYPE* bytes, const size_t length);
```

Comment:
Files using C++ features (e.g. templates) should be named `.hpp` instead of `.h`. This is because these files are compiled only in C++.

Here are template methods and thus it should be `.hpp`.

Reply:
Understood.
Since we were using `ha_lineairdb.h`, I thought it was ok to use `.h` for c++ header files.